### PR TITLE
Migrate to flaggy

### PR DESCRIPTION
**Pull request by mcintyre94**
As discussed in https://github.com/jesseduffield/lazydocker/issues/11, migrating from Flag to Flaggy is a prerequisite for supporting slice arguments. This PR moves the existing code from Flag to Flaggy without any changes in behaviour or additional flags being added yet.

One thing to note is that Flaggy has a built-in version flag, which I'm using with the previous version output.

I've also added a Name/Description/Repo link to the help output - those are in a second commit in case they're not wanted/you want them changed.

New help output:

```
$ lazydocker --help
lazydocker - The lazier way to manage everything docker
https://github.com/jesseduffield/lazydocker

  Flags:
       --version  Displays the program version string.
    -h --help  Displays help with available flag, subcommand, and positional value parameters.
    -config --config  Print the current default config
    -d --debug  a boolean
```

Version output (on my machine):

```
$ lazydocker --version
Version: commit=, build date=, build source=unknown, version=unversioned, os=darwin, arch=amd64
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -21,29 +21,33 @@ var (
...
-		fmt.Printf("commit=%s, build date=%s, build source=%s, version=%s, os=%s, arch=%s\n", commit, date, buildSource, version, runtime.GOOS, runtime.GOARCH)
-		os.Exit(0)
-	}
 
-	if *configFlag {
+	flaggy.SetName("lazydocker")
+	flaggy.SetDescription("The lazier way to manage everything docker")
+	flaggy.DefaultParser.AdditionalHelpPrepend = "https://github.com/jesseduffield/lazydocker"
+
+	flaggy.Bool(&configFlag, "config", "config", "Print the current default config")
```
Comment (dawidd6) [inline main.go]: I think that the shorthand for `config` should be `c`.

Comment (mcintyre94) [reply main.go]: Oops, thankyou!

**Comment by dawidd6** [comment]
Sooo, are you going to add more flags, like `-f` within this pull request or will you make another one for this?

**Comment by mcintyre94** [comment]
@dawidd6 If you're happy with these changes, shall I squash into one commit?

**Comment by dawidd6** [comment]
Sure, can't see why not.

File: main.go [hunk 2]
```diff
@@ -21,29 +21,33 @@ var (
...
+	if configFlag {
 		var buf bytes.Buffer
 		yaml.NewEncoder(&buf).Encode(config.GetDefaultConfig())
 		fmt.Printf("%v\n", buf.String())
 		os.Exit(0)
 	}
 
 	// for now we're always in debug mode so we're not passing *debuggingFlag
-	*debuggingFlag = true
+	debuggingFlag = true
```
Comment (jesseduffield) [inline main.go]: we can actually remove lines 47 and 48 because that was only there for my own personal convenience :P

Comment (mcintyre94) [reply main.go]: Good shout! :)

**Comment by mcintyre94** [comment]
@jesseduffield Newbie go question! I ran `dep ensure`, and it also added `vendor/github.com/jesseduffield/lazydocker/` - is there any way to avoid that happening in my fork? I figure we don't want it here :)

**Comment by dawidd6** [comment]
@mcintyre94 did you change some imports to point to your fork?

**Comment by mcintyre94** [comment]
> @mcintyre94 did you change some imports to point to your fork?

Not that I can see - `git grep mcintyre94` is empty, and the only diff to what's currently pushed here is that I ran `dep ensure` so no import changes that aren't pushed.

**Comment by mcintyre94** [comment]
I can remove it before I push, but just wondered if there was something I could do to avoid that happening next time :)

**Comment by jesseduffield** [comment]
@mcintyre94 I'm actually not sure how to get around that issue. It might be a limitation of using dep with forks :P

**Comment by jesseduffield** [comment]
At any rate, I'm happy with this PR so it's merge time!

**Pull request merged**